### DISCUSSION TITLE
How to install formula with compile option?

### DISCUSSION THREAD

**Discussion by scapodicasa** [Homebrew/discussions#48] [Everyday usage]
Seeing that Formulae does not support options, I need to install a formula that in past had a compiling options (`haproxy` with lua).

I can create a custom formula with the desired option and install it, no problem. What about contributing to Homebrew giving this option to other people? What's the right way? Can I create a pull request with a new formula (something like `haproxy-lua`) or something else?

Sorry if this is a topic already discussed in past, but I cannot find answer to this on documentation.

**Comment by SMillerDev** [comment]
https://docs.brew.sh/Formula-Cookbook#specifying-other-formulae-as-dependencies

> Note: options are not allowed in Homebrew/homebrew-core as they are not tested by CI.

**Reply by scapodicasa** [reply]
Yes, I read it, but so can I create a pull request of a new formula that includes the option (not as an option, but as a regular dependency) that I (and, maybe other people) want to use? Or this types of formula are not allowed? In case they are not allowed, what is the best way?

**Reply by SMillerDev** [reply]
If you think the option is useful for the majority of the users of the formula, just add it to the existing one. If you don't think it will you'll have to keep it in a tap.
